dev-docs/RFCs/vNext/composite-layer-customization-rfc.md

```diff
@@ -40,62 +40,61 @@ const customizedGeoJsonLayer = new GeoJsonLayer({
   ...,
   transferAllProps: true,
   subLayerProps: {
-    ScatterplotLayer: {
-      layer: ExtendedScatterplotLayer,
+    points: {
+      type: ExtendedScatterplotLayer,
       extendedProp: 100
     },
-    PolygonLayer: {
+    'polygons-fill': {
       subLayerProps: {  // Recursive application of sublayerProps!
         SolidPolygonLayer: {
-          layer: HollowPolygonLayer
+          type: HollowPolygonLayer
         }
       }
     }
   }
 })
 ```
 
-In geojson:
-```js
-const defaultProps = {
-  ...,
-  subLayers: {
-    PointLayer: ScatterplotLayer,
-    LineLayer: PathLayer,
-    PolygonLayer: SolidPolygonLayer
-  }
-};
-
-const GeoJsonLayer extends CompositeLayer {
-  ...
-  renderLayers() {
-  	return [
-  	  new this.props.subLayerProps.PointLayer({
-  	  }),
-  	];
-  }
-}
-```
+#### Changes to the CompositeLayer API
 
+**getSubLayerProps**
 
-### Proposal: CompositeLayers take a `subLayerProps` prop
+The overriding props supplied by the user can be merged inside `compositeLayer.getSubLayerProps` so that each layer does not have to implement their own handling of `subLayerProps`.
 
+**getSubLayerClass**
+
+A new CompositeLayer class method that extracts the `type` field from `subLayerProps`, if any. This will replace the ad-hoc implementations in e.g. [HexagonLayer](https://github.com/uber/deck.gl/blob/6.3-release/modules/layers/src/hexagon-layer/hexagon-layer.js#L396) and [GeoJsonLayer](https://github.com/uber/deck.gl/blob/6.3-release/modules/layers/src/geojson-layer/geojson-layer.js#L71).
 
-In application:
-```js
-  const customizedGeoJsonLayer = new GeoJsonLayer({
-  	...,
-  	subLayerProps: {
-  	  PointLayer: ...
-  	}
-  })
-```
 
+#### Open Questions
 
+**Should keys be layer class name or sublayer id?**
+
+Composite layers may render multiple sublayers of the same type. For example, the GeoJsonLayer renders two `PathLayer` instances, one for polygon outlines and one for line strings. Differentiating them with unique ids would give the user finer grain control.
+
+**How do user props merge with props created by the parent layer?**
+
+The composite layer themselves should decide which props are "overridable" by the user. This can be done via supplying multiple arguments to the sublayer constructor:
```

Comment:
Is there a way for the user to know which `props` are "overridable" and which are not?

Reply:
These will need to be documented on a case-by-case basis

Reply:
Updated the section to clarify: 1) default behavior (external always overrides internal) 2) generic recommendation (all props are overridable) 3) mechanism to block override and when to use it